app/controllers/staff/speakers_controller.rb

```diff
@@ -1,60 +1,51 @@
 class Staff::SpeakersController < Staff::ApplicationController
   decorates_assigned :speaker
-  before_filter :set_proposal, only: [:new, :create, :destroy]
+  before_action :set_program_session, only: [:new, :create]
 
   def index
-    render locals: {
-             proposals: @event.proposals.includes(speakers: :user).decorate
-           }
+    @program_speakers = current_event.speakers.in_program
   end
 
   def new
     @speaker = Speaker.new
-    @user = @speaker.build_user
-    @proposal = Proposal.find_by(uuid: params[:proposal_uuid])
   end
 
   def create
     s_params = speaker_params
-    user = User.find_by(email: s_params.delete(:speaker_email))
-    if user
-      if user.speakers.create(s_params.merge(proposal: @proposal))
-        flash[:success] = "Speaker was added to this proposal"
-      else
-        flash[:danger] = "There was a problem saving this speaker"
-      end
+    @speaker = @program_session.speakers.create(s_params.merge(event: current_event))
```

Comment:
I think we should use pundit to authorize the user for CRUD actions. Should be organizer-only.